src/page/EditCoursePage.jsx

```diff
@@ -1,55 +1,52 @@
 import React, { Component } from 'react';
 import { AppContainer } from '../components/app-container';
 import { EditCourse } from '../components/edit-course';
-// import { getAll } from "../api/CoursesApi";
+import { connect } from 'react-redux';
+import { getById, updateField, update, clean } from '../store/actions/courses';
 import { withRouter } from "react-router-dom";
-// import axios from 'axios';
 
 export class EditCoursePageContainer extends Component {
-    constructor(props) {
-        super(props);
-
-        this.state = {
-            course: {},
-            possibleAuthors: [],
-            courseAuthors: []
-        };
-    }
 
     componentDidMount() {
-        const possibleAuthors = [];
-        const courseAuthors = [];
-        this.setState({
-            possibleAuthors,
-            courseAuthors
-        })
+        this.props.getById(this.props.match.params.id);
     }
+
     handleChange = (event, field) => {
-        this.setState({
-            ...this.state,
-            course: {
-                ...this.state.course,
-                [field]: event.target.value
-            }
-        });
+        this.props.updateField(field, event.target.value);
     }
 
     handleSave = () => {
-        // updateCourse(this.state.course);
+        this.props.update(this.props.course);
+        this.props.history.push(`/courses`);
+    }
+
+    handleCancle = () => {
         this.props.history.push(`/courses`);
     }
+
+    componentWillUnmount() {
+        this.props.clean();
+    }
+
     render() {
         return (
             <AppContainer>
                 <EditCourse
-                    course={this.state.course}
+                    course={this.props.course}
                     onChange={this.handleChange}
                     onSave={this.handleSave}
-                    possibleAuthors={this.state.possibleAuthors}
-                    courseAuthors={this.state.courseAuthors} />
+                    onCancle={this.handleCancle}
+                    possibleAuthors={[]}
+                    courseAuthors={[]} />
             </AppContainer>
         );
     }
 }
 
-export const EditCoursePage = withRouter(EditCoursePageContainer);
\ No newline at end of file
+const mapStateToProps = (state) => {
+    return {
+        course: state.course.courseEntity
```

Comment:
again i see `state.course.courseEntity` that why we should keep that logic in selectors near it's store

Reply:
fixed